### PR TITLE
Adds the ability to regenerate service files.

### PR DESCRIPTION
**Pull request by JeffreyDevloo**

### CODE REVIEW

File: ovs/extensions/services/systemd.py [hunk 1]
```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
...
+                    startup_dependency.split('.')[:-1])  # Remove .service from startup dependency
+            output = Systemd.add_service(name=name,
+                                         client=client,
+                                         params=service_params,
+                                         target_name=target_name,
+                                         startup_dependency=startup_dependency,
+                                         delay_registration=True)
+            if output is None:
+                raise RuntimeError('Regenerating files for service {0} has failed'.format(target_name))
+            else:
```
Comment (khenderick) [inline ovs/extensions/services/systemd.py]: Just `return True` here instead of the `else`-branch

**Review by khenderick — CHANGES_REQUESTED** [review]

File: ovs/extensions/services/systemd.py [hunk 2]
```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
         return params
 
     @staticmethod
+    def regenerate_service(name, client, target_name):
+        """
```
Comment (kvanhijf) [inline ovs/extensions/services/systemd.py:110]: What exactly does this method?

File: ovs/extensions/services/systemd.py [hunk 3]
```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
...
+        :param name: Template name of the service to add
```
Comment (kvanhijf) [inline ovs/extensions/services/systemd.py]: You don't add a service

File: ovs/extensions/services/systemd.py [hunk 4]
```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
...
+        :type name: str
+        :param client: Client on which to add the service
+        :type client: ovs.extensions.generic.sshclient.SSHClient
+        :param target_name: The current service name eg ovs-volumedriver_flash01.service
+        :type target_name: str
+        :return: True if successful
```
Comment (kvanhijf) [inline ovs/extensions/services/systemd.py]: Shouldn't return anything imo. Raise if anything goes wrong, None in successful scenario

File: ovs/extensions/services/systemd.py [hunk 5]
```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
...
+        :rtype bool
+        """
+        configuration_key = '/ovs/framework/hosts/{0}/services/{1}'.format(System.get_my_machine_id(client),
```
Comment (kvanhijf) [inline ovs/extensions/services/systemd.py]: The key `/ovs/framework/hosts...` should be a class variable

File: ovs/extensions/services/systemd.py [hunk 6]
```diff
@@ -105,6 +105,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
...
+                                                                           target_name.replace('ovs-', ''))
+        # If the entry is stored in arakoon, it means the service file was previously made
+        if Configuration.exists(configuration_key):
```
Comment (kvanhijf) [inline ovs/extensions/services/systemd.py]: If key does not exist, an error should be thrown mentioning `add_service` should be used

File: ovs/extensions/services/upstart.py [hunk 7]
```diff
@@ -98,6 +98,41 @@ def add_service(name, client, params=None, target_name=None, startup_dependency=
         return params
 
     @staticmethod
+    def regenerate_service(name, client, target_name):
+        """
+        :param name: Template name of the service to add
```
Comment (kvanhijf) [inline ovs/extensions/services/upstart.py]: Identical remarks as for systemd

**Review by kvanhijf — CHANGES_REQUESTED** [review]

**Pull request closed without merging**